config/options.go

```diff
@@ -718,22 +716,32 @@ func (o *Options) GetAuthenticateURL() *url.URL {
 	return u
 }
 
-// GetAuthorizeURL returns the AuthorizeURL in the options or 127.0.0.1:5443.
-func (o *Options) GetAuthorizeURL() *url.URL {
-	if o != nil && o.AuthorizeURL != nil {
-		return o.AuthorizeURL
-	}
-	u, _ := url.Parse("http://127.0.0.1" + DefaultAlternativeAddr)
-	return u
+// GetAuthorizeURLs returns the AuthorizeURLs in the options or 127.0.0.1:5443.
+func (o *Options) GetAuthorizeURLs() []*url.URL {
+	return o.getURLs(append([]string{o.AuthorizeURLString}, o.AuthorizeURLStrings...)...)
 }
 
-// GetDataBrokerURL returns the DataBrokerURL in the options or 127.0.0.1:5443.
-func (o *Options) GetDataBrokerURL() *url.URL {
-	if o != nil && o.DataBrokerURL != nil {
-		return o.DataBrokerURL
+// GetDataBrokerURLs returns the DataBrokerURLs in the options or 127.0.0.1:5443.
+func (o *Options) GetDataBrokerURLs() []*url.URL {
+	return o.getURLs(append([]string{o.DataBrokerURLString}, o.DataBrokerURLStrings...)...)
+}
+
+func (o *Options) getURLs(strs ...string) []*url.URL {
+	var urls []*url.URL
+	if o != nil {
+		for _, str := range strs {
+			u, err := urlutil.ParseAndValidateURL(str)
+			if err == nil {
+				urls = append(urls, u)
+			}
```

Comment:
error has to be propagated up

Reply:
Error handling was done in Validate. Returning an error will complicate code downstream. Do you still want me to do this?

Reply:
in options.Validate there's only check for DataBrokerUrlString, not for a slice https://github.com/pomerium/pomerium/blob/1a33d753c24d6e2bdc677e80235712b7cd47dd83/config/options.go#L577

Reply:
ok I will make the changes as requested.

Reply:
I attempted to make these changes and as expected this will modify several hundred lines of code and touch most of the existing XDS code, as well as all of our tests. I am not comfortable making such a change in this PR. I will close this PR and start over.